abstractions/python/kiota/abstractions/serialization/serialization_writer.py

```diff
@@ -0,0 +1,270 @@
+
+from abc import ABC, abstractmethod
+from datetime import date, datetime, time, timedelta
+from enum import Enum
+from io import BytesIO
+from typing import Any, Callable, Dict, List, Optional, TypeVar
+from uuid import UUID
+
+from .parsable import Parsable
+
+T = TypeVar("T")
+U = TypeVar("U", bound=Parsable)
+
+
+class SerializationWriter(ABC):
+    """Defines an interface for serialization of objects to a stream
+    """
+    @abstractmethod
+    def write_string_value(self, key: Optional[str], value: Optional[str]) -> None:
+        """Writes the specified string value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[str]): The string value to be written.
+        """
+        pass
+
+    @abstractmethod
+    def write_boolean_value(self, key: Optional[str], value: Optional[bool]) -> None:
+        """Writes the specified boolean value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[bool]): The boolean value to be written.
+        """
+        pass
+
+    @abstractmethod
+    def write_int_value(self, key: Optional[str], value: Optional[int]) -> None:
+        """Writes the specified integer value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[int]): The integer value to be written.
+        """
+        pass
+
+    @abstractmethod
+    def write_float_value(self, key: Optional[str], value: Optional[float]) -> None:
+        """Writes the specified float value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[float]): The float value to be written.
+        """
+        pass
+
+    @abstractmethod
+    def write_uuid_value(self, key: Optional[str], value: Optional[UUID]) -> None:
+        """Writes the specified uuid value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[UUId]): The uuid value to be written.
+        """
+        pass
+
+    @abstractmethod
+    def write_datetime_offset_value(self, key: Optional[str], value: Optional[datetime]) -> None:
+        """Writes the specified datetime offset value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[datetime]): The datetime offset value to be written.
+        """
+        pass
+
+    @abstractmethod
+    def write_timespan_value(self, key: Optional[str], value: Optional[timedelta]) -> None:
```

Comment:
`write_timedelta_value`? (same as parse_node)
That's assuming Duration types will be mapped to timedelta in Python

Reply:
Yes, time durations are represented by `timedelta` values in python. Renamed the methods in https://github.com/microsoft/kiota/pull/1058/commits/016eadd4a8a75068bb612bd9025bc7adb2b9ae7b